test/src/org/apache/jmeter/protocol/http/util/TestHTTPArgument.java

```diff
@@ -22,10 +22,15 @@
 
 import org.apache.jmeter.config.Arguments;
 import org.apache.jmeter.testelement.property.CollectionProperty;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 public class TestHTTPArgument {
 
+        @Rule
```

Comment:
It tried your test class without touching the existing code, it does not show existing issue.
The failures are due to changes in contract (Switch to Error instead of IllegalArgumentException)

 There should be a dedicated test showing broken code and once we apply the fix, test should succeed using Shift_JS encoding for example.